linera-core/src/chain_worker/state/temporary_changes.rs

```diff
@@ -213,6 +215,29 @@ where
         for blob in blobs {
             self.0.cache_recent_blob(Cow::Borrowed(blob)).await;
         }
+        for blob in self.0.get_blobs(block.published_blob_ids()).await? {
+            let size = match blob.content() {
+                BlobContent::Data(bytes) => bytes.len(),
+                BlobContent::ContractBytecode(compressed_bytecode)
+                | BlobContent::ServiceBytecode(compressed_bytecode) => {
+                    compressed_bytecode
+                        .decompress(maximum_bytecode_size)
+                        .map_err(|error| {
+                            ChainError::ExecutionError(
+                                ExecutionError::from(error),
+                                linera_chain::ChainExecutionContext::Block,
+                            )
+                        })?;
+                    compressed_bytecode.compressed_bytes.len()
```

Comment:
No: This is for the _other_ check, against `maximum_blob_size`. That one applies to the _compressed_ size, because that's how big the blob is.

The check against `maximum_bytecode_size` was already done; it applies to the _uncompressed_ bytecode.

Reply:
I see. By trying to be "efficient" we lost the clarity.

Reply:
I'll actually change `decompress` back to what it was, and add a function that just checks the size. Maybe then it will be clearer.

Reply:
Why do we verify blob size against compressed variant? is it to not bloat blocks too much? b/c once decompressed, blobs are stored in their decompressed version on chain?

Reply:
In storage and in transit they are always compressed, so this limit remains as relevant for bytecode as for other blobs.

The bytecode limit is to defend against zip bombs, and to limit _memory_ usage.

Reply:
OK. Just to explain myself - I was thinking about these in context of "application as blob" and in this case we'd first decompress and then "install" the app. I guess we still store the compressed version for the sake of block's content itself. Is "block storage" separate from "app storage"?

Reply:
No, I think currently the bytecode is only ever stored in compressed form, and only decompressed in memory. On busy chains, the chain worker will remain in memory, together with the decompressed bytecode, so it doesn't have to be decompressed in each block.

Reply:
Thanks.